ecosystem/sf-indexer/aptos-substreams/src/token_converter.rs

```diff
@@ -0,0 +1,369 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::{Context, Result};
+use aptos_protos::{
+    block_output::v1::{MoveResourceOutput, TableItemOutput},
+    tokens::v1::{CollectionData, Token, TokenData, TokenDataId, TokenId},
+};
+use std::collections::HashMap;
+
+/// Mapping from table handle to owner type, including type of the table (AKA resource type)
+pub fn get_table_handle_to_owner(
+    resource: &MoveResourceOutput,
+    txn_version: u64,
+) -> anyhow::Result<Option<HashMap<String, (String, String)>>> {
+    match resource.type_str.as_str() {
+        "0x3::token::Collections" => {
+            let address = &resource.address;
+            let data: serde_json::Value = serde_json::from_str(&resource.data)?;
+            let collection_handle = data["data"]["collection_data"]["handle"]
+                .as_str()
+                .map(|s| s.to_string())
+                .context(format!(
+                    "version {} failed! collection data handle must be present {:?}",
+                    txn_version, data
+                ))?;
+
+            Ok(Some(HashMap::from([(
+                standardize_handle(&collection_handle),
+                (address.clone(), resource.type_str.clone()),
+            )])))
```

Comment:
Can you make a struct for the 2nd part?

Also have a type alias for the first one?

Reply:
sure.